src/test/java/org/sonatype/nexus/ci/iq/ClassFilterLoggingTestTrait.groovy

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package org.sonatype.nexus.ci.iq
+
+import java.util.logging.Level
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowExecution
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.jvnet.hudson.test.LoggerRule
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Add this trait to any test that should have an extra verification that no JEP-200 marshalling errors have occurred.
+ * See INT-2407 for additional information and INT-418 for original work.
+ */
+trait ClassFilterLoggingTestTrait
+{
+  Logger logger = LoggerFactory.getLogger(ClassFilterLoggingTestTrait)
+
+  @Rule
+  public LoggerRule loggerRule = new LoggerRule()
+
+  def helpfulMessage = '''A possible JEP-200 marshalling error has occurred. It is likely that additional classes 
+will need to be added to the 'hudson.remoting.ClassFilter' file. See INT-2407 for additional info'''
```

Comment:
might a dedicated wiki page be good to have, since it's easy for the problem to come up repeatedly? A jira number is nicely concise, though. 👍

Reply:
Good idea! https://docs.sonatype.com/display/INT/Jenkins+Plugin+and+JEP-200